src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1273,33 +1312,103 @@ private int GetMetadataIndex(string value)
                 return index;
             }
 
-            private bool ShouldCopyLocalPackageAssets(LockFileTargetLibrary package)
+            private void ComputePackageExclusions()
             {
-                return _platformPackageExclusions == null || !_platformPackageExclusions.Contains(package.Name);
-            }
-
-            private HashSet<string> GetPlatformPackageExclusions()
-            {
-                // Only exclude packages for framework-dependent applications
-                if (_task.IsSelfContained && !string.IsNullOrEmpty(_runtimeTarget.RuntimeIdentifier))
-                {
-                    return null;
-                }
-
-                var packageExclusions = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                var copyLocalPackageExclusions = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
                 var libraryLookup = _runtimeTarget.Libraries.ToDictionary(p => p.Name, StringComparer.OrdinalIgnoreCase);
 
-                // Exclude the platform library
-                if (_task.PlatformLibraryName != null)
+                // Only exclude platform packages for framework-dependent applications
+                if ((!_task.IsSelfContained || string.IsNullOrEmpty(_runtimeTarget.RuntimeIdentifier)) &&
+                    _task.PlatformLibraryName != null)
                 {
+                    // Exclude the platform library
                     var platformLibrary = _runtimeTarget.GetLibrary(_task.PlatformLibraryName);
                     if (platformLibrary != null)
                     {
-                        packageExclusions.UnionWith(_runtimeTarget.GetPlatformExclusionList(platformLibrary, libraryLookup));
+                        copyLocalPackageExclusions.UnionWith(_runtimeTarget.GetPlatformExclusionList(platformLibrary, libraryLookup));
+
+                        // If the platform library is not Microsoft.NETCore.App, treat it as an implicit dependency.
```

Comment:
Why don't I see this being removed from:

https://github.com/dotnet/sdk/blob/8cb31b0fd0bb5d36b9781442865f2e26cf6fbedc/src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs#L102

Do we need both, and why? I was expecting to see things moved in to here from elsewhere. Is that the right way to think about this change?

Reply:
I think the duplicate logic is still used by `ResolveCopyLocalAssets` (the old publish asset resolving code that still gets invoked when there are runtime stores in use after these changes).

We might be able to consolidate this bit, but might also not be worth the effort if the intent is to remove `ResolveCopyLocalAssets` and the `AssetsFileResolver` (and then by extension, `ProjectContext.GetRuntimeLibraries` as referenced).

Reply:
Yes, the code in `ProjectContext` is still used by `ResolveCopyLocalAssets`.

My hope is that we can remove `ResolveCopyLocalAssets` with further refactoring.  This would require supporting the equivalent of PreserveStoreLayout and RuntimeStorePackages in `ResolvePackageAssets`.